src/buildpack.ts

```diff
@@ -60,10 +70,68 @@ export async function getDockerfile(b: Builder): Promise<string> {
       USER deno
       ENTRYPOINT ["deno", "run", "-A", "${b.builderConfig.entrypoint}"]
     `;
+  } else if (b.builder === "node-typescript") {
+    // Builder runs node Docker image, installs using npm (if package-lock.json) else yarn, then compiles using tsc
+    const { entrypoint } = b.builderConfig;
+    // Find the closest directory to entrypoint as working directory
+    let workingDir = null;
+    const pathParts = dirname(entrypoint).split(sep);
+    while (pathParts.length >= 0) {
+      if (existsSync(join(...pathParts, "package.json"))) {
+        workingDir = join(...pathParts);
+        break;
+      }
+      pathParts.pop();
+    }
+    if (workingDir === null) {
+      throw new Error(
+        `Could not find package.json in any directories above ${b.builderConfig.entrypoint}`
+      );
+    }
```

Comment:
My last PR actually just added a helper for this

```suggestion
const workingDir = find("package.json", dirname(entrypoint))
if (!workingDir) {...}
```